### PR TITLE
Remove persistent task when AllocatedTask fails during init

### PR DESCRIPTION
**Pull request by polyfractal**
If a task encounters an issue while initializing, the allocated task is correctly removed from the running tasks and a warning is logged.  But the persistent task remains in the cluster state, which leads to a "zombie" persistent task that cannot be removed. The allocated task is gone so it can't be canceled, and if the persistent task is re-allocated it is likely to fail again because the issue happened during initialization.

This adjusts the error handling to also inform the master that the task failed, so that it may be removed from the CS

I wasn't really sure what exception to throw, so just used a RuntimeException.  Happy to change it to something more appropriate.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java [hunk 1]
```diff
@@ -197,9 +197,11 @@ public Task createTask(long id, String type, String action, TaskId parentTaskId,
         } finally {
             if (processed == false) {
                 // something went wrong - unregistering task
-                logger.warn("Persistent task [{}] with id [{}] and allocation id [{}] failed to create", task.getAction(),
-                        task.getPersistentTaskId(), task.getAllocationId());
+                ParameterizedMessage e = new ParameterizedMessage("Persistent task [{}] with id [{}] and allocation id " +
+                    "[{}] failed to create", task.getAction(), task.getPersistentTaskId(), task.getAllocationId());
+                logger.warn(e.getFormattedMessage());
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java:202]: why did you change the logging here? AFAICS that change was unnecessary

**Review by ywelsch — CHANGES_REQUESTED** [review]
Can you add a test as well?

File: server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java [hunk 2]
```diff
@@ -197,9 +197,11 @@ public Task createTask(long id, String type, String action, TaskId parentTaskId,
...
                 taskManager.unregister(task);
+                notifyMasterOfFailedTask(taskInProgress, new RuntimeException(e.getFormattedMessage()));
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java:204]: we should rather capture the exception thrown and use that here

File: server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java (hunk 1 above)
Comment (polyfractal) [reply server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java:202]: It's the same message, just reusing it for the logger and the exception so it's not c/p.  Can just write it out twice if you prefer.

File: server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java (hunk 2 above)
Comment (polyfractal) [reply server/src/main/java/org/elasticsearch/persistent/PersistentTasksNodeService.java:204]: I'll see what I can do. The setup of this area is tricky since it's relying on `processed` flag and a finally clause rather than exception catching.  Was trying to keep changes minimal, but I think I can stash the caught exception and use it in the finally clause.

**Comment by ywelsch** [comment]
@polyfractal are you still interested in picking this up?

**Comment by polyfractal** [comment]
Sorry this fell through the cracks.  I'm not sure when I'll have time to work on this... I think it's important though, so if someone has bandwidth and wants to take it please feel free. :)

**Comment by henningandersen** [comment]
Superceded by #71462
